lib/constraints/regex.ex

```diff
@@ -0,0 +1,15 @@
+defmodule AbsintheHelpers.Constraints.Regex do
+  @moduledoc """
+  Applies regex constraint on node data. This constraint can only be applied to String types.
+  """
+
+  @behaviour AbsintheHelpers.Constraint
+
+  def call(node = %{data: data}, {:regex, regex}) do
+    if data =~ Regex.compile!(regex) do
```

Comment:
This will compile the regex on every request - you probably want to do it only once when defining

Reply:
I've tried that actually, but Absinthe somehow [tries to map it](https://github.com/absinthe-graphql/absinthe/blob/e37e2858c330bda08cbdb1d62f7c6fa7b3a1ca32/lib/absinthe/blueprint/input.ex#L72-L82) since it's a struct

If I have this
```elixir
        field(:description, :string) do
          directive(:constraints, regex: ~r/^[a-zA-Z\s]+$/, min: 5, max: 50)
        end
```

then I get this
```elixir
== Compilation error in file test/absinthe_helpers/phases/apply_constraints_test.exs ==
** (Protocol.UndefinedError) protocol Enumerable not implemented for ~r/^[a-zA-Z\s]+$/ of type Regex (a struct). This protocol is implemented for the following type(s): Date.Range, File.Stream, Function, GenEvent.Stream, HashDict, HashSet, IO.Stream, Jason.OrderedObject, List, Map, MapSet, Range, Stream
    (elixir 1.16.2) lib/enum.ex:1: Enumerable.impl_for!/1
    (elixir 1.16.2) lib/enum.ex:166: Enumerable.reduce/3
    (elixir 1.16.2) lib/enum.ex:4396: Enum.map/2
    (absinthe 1.7.8) lib/absinthe/blueprint/input.ex:75: Absinthe.Blueprint.Input.parse/1
    (absinthe 1.7.8) lib/absinthe/blueprint/input/value.ex:47: Absinthe.Blueprint.Input.Value.build/1
    (absinthe 1.7.8) lib/absinthe/schema/notation.ex:1851: anonymous fn/2 in Absinthe.Schema.Notation.build_directive_arguments/2
    (elixir 1.16.2) lib/enum.ex:1700: Enum."-map/2-lists^map/1-1-"/2
    (absinthe 1.7.8) lib/absinthe/schema/notation.ex:1845: Absinthe.Schema.Notation.build_directive_arguments/2
    (absinthe 1.7.8) lib/absinthe/schema/notation.ex:1833: Absinthe.Schema.Notation.record_applied_directive!/3
    (absinthe 1.7.8) expanding macro: Absinthe.Schema.Notation.directive/2
    test/absinthe_helpers/phases/apply_constraints_test.exs:34: AbsintheHelpers.Phases.ApplyConstraintsTest.TestSchema (module)
    (absinthe 1.7.8) expanding macro: Absinthe.Schema.Notation.field/3
    test/absinthe_helpers/phases/apply_constraints_test.exs:33: AbsintheHelpers.Phases.ApplyConstraintsTest.TestSchema (module)
    (absinthe 1.7.8) expanding macro: Absinthe.Schema.Notation.input_object/2
    test/absinthe_helpers/phases/apply_constraints_test.exs:30: AbsintheHelpers.Phases.ApplyConstraintsTest.TestSchema (module)
```

```elixir
iex(26)> Enumerable.impl_for ~r//
nil
```

Not sure how we could work around this 🤔 or whether there's a better approach.